packages/graphrag-common/graphrag_common/config/load_config.py

```diff
@@ -0,0 +1,179 @@
+# Copyright (c) 2024 Microsoft Corporation.
+# Licensed under the MIT License
+
+"""Load configuration."""
+
+import json
+import os
+from collections.abc import Callable
+from pathlib import Path
+from string import Template
+from typing import Any, TypeVar
+
+import yaml
+from dotenv import load_dotenv
+
+T = TypeVar("T", covariant=True)
+
+
+class ConfigParsingError(ValueError):
+    """Configuration Parsing Error."""
+
+    def __init__(self, msg: str) -> None:
+        """Initialize the ConfigParsingError."""
+        super().__init__(msg)
+
+
+def _parse_json(data: str) -> dict[str, Any]:
+    """Parse JSON data."""
+    return json.loads(data)
+
+
+def _parse_yaml(data: str) -> dict[str, Any]:
+    """Parse YAML data."""
+    return yaml.safe_load(data)
+
+
+def _get_parser_for_file(file_path: str | Path) -> Callable[[str], dict[str, Any]]:
+    """Get the parser for the given file path."""
+    file_path = Path(file_path).resolve()
+    match file_path.suffix.lower():
+        case ".json":
+            return _parse_json
+        case ".yaml" | ".yml":
+            return _parse_yaml
+        case _:
+            msg = (
+                f"Failed to parse, {file_path}. Unsupported file extension, "
+                + f"{file_path.suffix}. Pass in a custom config_parser argument or "
+                + "use one of the supported file extensions, .json, .yaml, .yml, .toml."
+            )
+            raise ConfigParsingError(msg)
+
+
+def _load_dotenv(env_file_path: Path | str) -> None:
+    """Load the .env file if it exists."""
+    env_file_path = Path(env_file_path).resolve()
+    if not env_file_path.is_file():
+        msg = f"dot_env_path not found: {env_file_path}"
+        raise FileNotFoundError(msg)
+    load_dotenv(env_file_path)
+
+
+def _parse_env_variables(text: str) -> str:
+    """Parse environment variables in the configuration text."""
+    try:
+        return Template(text).substitute(os.environ)
+    except KeyError as error:
+        msg = f"Environment variable not found: {error}"
+        raise ConfigParsingError(msg) from error
+
+
+def _recursive_merge_dicts(dest: dict[str, Any], src: dict[str, Any]) -> None:
+    """Recursively merge two dictionaries in place."""
+    for key, value in src.items():
+        if isinstance(value, dict):
+            if isinstance(dest.get(key), dict):
+                _recursive_merge_dicts(dest[key], value)
+            else:
+                dest[key] = value
+        else:
+            dest[key] = value
+
+
+def load_config(
+    config_initializer: Callable[..., T],
+    config_path: str | Path,
```

Comment:
Should this be named `config_filename` to make it more clear it is not a folder?